packages/eas-cli/src/run/android/adb.ts

```diff
@@ -0,0 +1,139 @@
+import spawnAsync, { SpawnResult } from '@expo/spawn-async';
+import os from 'os';
+
+import Log from '../../log';
+import { sleepAsync } from '../../utils/promise';
+import { sdkRoot } from './sdk';
+
+export interface AndroidEmulator {
+  pid?: string;
+  name: string;
+}
+
+const BEGINNING_OF_ADB_ERROR_MESSAGE = 'error: ';
+
+export const adbExecutable = getAdbExecutable();
+
+export async function adbAsync(...args: string[]): Promise<SpawnResult> {
+  try {
+    return await spawnAsync(adbExecutable, args);
+  } catch (error: any) {
+    let errorMessage = (error.stderr || error.stdout || error.message).trim();
+    if (errorMessage.startsWith(BEGINNING_OF_ADB_ERROR_MESSAGE)) {
+      errorMessage = errorMessage.substring(BEGINNING_OF_ADB_ERROR_MESSAGE.length);
+    }
+    error.message = errorMessage;
+    throw error;
+  }
+}
+
+function getAdbExecutable(): string {
+  if (sdkRoot) {
+    return `${sdkRoot}/platform-tools/adb`;
+  }
+
+  Log.debug('Failed to resolve the Android SDK path, falling back to global adb executable');
+  return 'adb';
+}
+
+export function sanitizeAdbDeviceName(deviceName: string): string | undefined {
+  return deviceName
+    .trim()
+    .split(/[\r\n]+/)
+    .shift();
+}
+
+/**
+ * Return the Emulator name for an emulator ID, this can be used to determine if an emulator is booted.
+ *
+ * @param devicePid a value like `emulator-5554` from `abd devices`
+ */
+export async function getAdbNameForDeviceIdAsync(emulatorPid: string): Promise<string | null> {
+  const { stdout } = await adbAsync('-s', emulatorPid, 'emu', 'avd', 'name');
+
+  if (stdout.match(/could not connect to TCP port .*: Connection refused/)) {
+    // Can also occur when the emulator does not exist.
+    throw new Error(`Emulator not found: ${stdout}`);
+  }
+
+  return sanitizeAdbDeviceName(stdout) ?? null;
+}
+
+// TODO: This is very expensive for some operations.
+export async function getRunningEmulatorsAsync(): Promise<AndroidEmulator[]> {
+  const { stdout } = await adbAsync('devices', '-l');
+
+  const splitItems = stdout.trim().replace(/\n$/, '').split(os.EOL);
+  // First line is `"List of devices attached"`, remove it
+  // @ts-ignore: todo
+  const attachedDevices: {
+    pid: string;
+    type: string;
+  }[] = splitItems
+    .slice(1, splitItems.length)
+    .map(line => {
+      // unauthorized: ['FA8251A00719', 'unauthorized', 'usb:338690048X', 'transport_id:5']
+      // authorized: ['FA8251A00719', 'device', 'usb:336592896X', 'product:walleye', 'model:Pixel_2', 'device:walleye', 'transport_id:4']
+      // emulator: ['emulator-5554', 'offline', 'transport_id:1']
+      const [pid] = line.split(' ').filter(Boolean);
```

Comment:
Search for the other `truthy` usages in the code.
```suggestion
      const [pid] = line.split(' ').filter(truthy);
```